src/WebJobs.Script.WebHost/Filters/AuthorizationLevelAttribute.cs

```diff
@@ -33,12 +33,20 @@ public AuthorizationLevelAttribute(AuthorizationLevel level)
                 throw new ArgumentNullException("actionContext");
             }
 
-            // determine the authorization level for the function and set it
-            // as a request property
-            var secretManager = actionContext.ControllerContext.Configuration.DependencyResolver.GetService<ISecretManager>();
+            AuthorizationLevel requestAuthorizationLevel = actionContext.Request.GetAuthorizationLevel();
+            
+            // If the request has not yet been authenticated, authenticate it
+            if (requestAuthorizationLevel == AuthorizationLevel.Anonymous)
```

Comment:
This line is the relevant change in the `AuthorizationLevelAttribute`, with this change, we'll only go through the authentication process (using `code` query string or `x-functions-key` header if the request is not yet authenticated.